GPflowOpt/design.py

```diff
@@ -19,18 +19,43 @@
 
 class Design(object):
     """
-    Space-filling designs generated within a domain.
+    Space-filling design of size N (number of points) generated within a D-dimensional domain.
+    
+    Users should call generate() which auto-scales the design to the domain specified in the constructor.
+    To implement new design methodologies subclasses should implement create_design(),
+    which returns the design on the domain specified by the generative_domain method (which defaults to a unit cube).
     """
 
     def __init__(self, size, domain):
         super(Design, self).__init__()
         self.size = size
         self.domain = domain
 
+    @property
+    def generative_domain(self):
+        """
+        :return: Domain object representing the domain associated with the points generated in create_design().
+        Defaults to [0,1]^D, can be overwritten by subclasses
+        """
+        return np.sum([ContinuousParameter('d{0}'.format(i), 0, 1) for i in np.arange(self.domain.size)])
+
     def generate(self):
         """
-        Returns a design of the requested size (N) and domain dimensionality (D). All data points are contained
-        by the design.
+        Returns a design transformed to the domain as specified in the constructor. All data points are in the
+        design specified in the constructor.
+        :return: 2D ndarray, N x D
+        """
+        Xs = self.create_design()
+        assert (Xs in self.generative_domain)
+        assert (Xs.shape == (self.size, self.domain.size))
+        transform = self.generative_domain >> self.domain
+        X = np.clip(transform.forward(Xs), self.domain.lower, self.domain.upper)
```

Comment:
Any reason we explicitly clip? If the designs are generated according to the self.generative_domain() all should be OK.

And since this is part of designs.py I would rather assert on it than clip them.